### DISCUSSION TITLE
Implemented custom domain feature for my SaaS

### DISCUSSION THREAD

**Discussion by damechen** [vercel/community#31] [Help] [answered]
Hey team,

I have a domain whose DNS is managed by Vercel. I setup wildcard subdomain for it.

Now I want to link my client's own subdomain to the subdomain of my saas site. 

Here are my questions:

1. Is there any limit on how many domains can be added to one project? In my case, I need to add my clients' site to my project.
2. How to manage the DNS so that my client's site can be pointed to my saas site?

Thanks!

**Comment by mcsdevv** [comment]
1. There is a limit of 50 domains per project for free accounts, unlimited for paid accounts (Pro and Enterprise).
2. The easiest way is to ask them to [add a CNAME record](https://vercel.com/docs/concepts/projects/custom-domains#subdomains).

**Comment by paulogdm** [comment]
**Update:** Happy to share there's now [unlimited custom domains](https://vercel.com/changelog/unlimited-custom-domains-for-all-pro-teams) on paid accounts (Pro and Enterprise). 

We've also released the [Platforms Starter Kit](https://demo.vercel.pub/platforms-starter-kit) to help you build your multi-tenant application with custom domains!

**Reply by Wtrapp** [reply]
Any update @paulogdm ?

**Reply by paulogdm** [reply]
Many! This feature should be released in a question of days. You should see it available in our changelog.

We will test it with selected customers before it becomes generally available.

**Reply by raimille1** [reply]
🤚 if you need volunteers @paulogdm

**Reply by paulogdm** [reply]
@raimille1 If you have a Customer Success Manager assigned to your account, please reach out to him as soon as possible :+1:

**Reply by adriano-tirloni** [reply]
@paulogdm Hey Paulo, do you now if this feature is going to become generally available in the next month?
If it doesn't we are going to implement our own alternative.

**Reply by Wtrapp** [reply]
@adriano-tirloni contact your account rep. We had to upgrade our plan and now we have more domains.

**Reply by lalit2005** [reply]
Any updates @paulogdm @leerob @rauchg @pacocoursey ??

**Reply by paulogdm** [reply]
This is now available to all customers with an Enterprise or tailored plan. Please reach out to our Sales department on sales@vercel.com if you need more than 50 domains on your team.

**Reply by ToledoNicola** [reply]
Hi, the guide is out?

**Reply by paulogdm** [reply]
Apologies, I had to delete my previous comment.
We are still working on the guide, and it should be out in the upcoming months.

**Reply by mdathersajjad** [reply]
I think to scale with 1000s of domain one can use a proxy caddy server which provides lot of functionality out of the box like auto ssl

**Reply by steven-tey** [reply]
Hey @Wtrapp @raimille1 @adriano-tirloni @Lalit2005 @ToledoNicola @mdathersajjad – I just posted the link to an example that should do exactly what you all are looking to do in my comment below! Sorry for the wait! 😅

**Comment by steven-tey** [comment]
Hey everyone! Great news – we recently rolled out a [simple example](https://github.com/vercel/examples/tree/main/edge-functions/hostname-rewrites) that allows you to create unique content pages with a multi-tenant infrastructure using [edge function](https://vercel.com/features/edge-functions) rewrites and [ISR](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration). Each of your users gets assigned a unique subdomain when they create their account, with the (usually paid) option to add a custom domain. We also helped [read.cv implement this](https://read.cv/support/Q00tMVbWBlA32mb9IKNn) for their site and it went really well!

We're also rolling out more complex boilerplates (e-commerce storefront builders, website builders) + supplementary code snippets (static tweets, domains API, OG image generator) in an [early access program](https://eap.vercel.sh/) – feel free to join if you're interested (click "sign up" on the Platforms on Vercel program).

**Reply by robotic-forest** [reply]
With this solution, will it be necessary to add custom domains to the “domains” tab in a project (which has the aforementioned 50 limit) or will this work with a greater (unlimited even?) number of domains, without having to add custom domains manually?

**Reply by lalit2005** [reply]
> With this solution, will it be necessary to add custom domains to the “domains” tab in a project (which has the aforementioned 50 limit) or will this work with a greater (unlimited even?) number of domains, without having to add custom domains manually?

I believe you should have an enterprise account on Vercel have more than 50 domains. Any service such as https://approximated.app should make having multiple different domains possible!

**Reply by steven-tey** [reply]
@robotic-forest @Lalit2005 Thanks for bringing this up! Right now what we're thinking is instead of charging based on custom domains, we will focus on charging based on bandwidth + serverless execution usage (GB/hrs). Therefore moving forward, we're planning to lift the limits on custom domains – especially for Platforms on Vercel customers like yourself!

Feel free to join our Platforms on Vercel early-access program here for more info: https://eap.vercel.sh/

**Comment by steven-tey** [comment, marked as answer]
Quick update on this – we recently launched the Platforms Starter Kit that's a comprehensive template for building multi-tenant applications with built-in custom domains support (+ unlimited custom domains 😉): https://demo.vercel.pub/platforms-starter-kit

Feel free to try it out and let us know what you think! :)

**Reply by steven-tey** [reply]
CC: @damechen @Wtrapp @adriano-tirloni @ToledoNicola @mdathersajjad @robotic-forest

**Reply by egoist** [reply]
Hi, the Vercel dashboard still indicates 50 as the domain limit for pro teams, will it be updated later or do you need to ask Vercel to explicitly update the limit?

**Reply by leerob** [reply]
Sorry about that, fixing the dashboard now - the limit is removed, you can ignore that.

https://vercel.com/changelog/unlimited-custom-domains-for-all-pro-teams
